### PR TITLE
Patrick Sandoval

### PR DESCRIPTION
**Pull request by patsand24**

### CODE REVIEW

File: security/ThreatModel.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### Threat Model
+
```
Comment (JCamp-12) [inline security/ThreatModel.md:2]: great e-mail, ty for following format!!

File: security/practical_security.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+So this prompted me to finally set-up 2-factor authentication on all my iphone and bank accounts.
+I had been putting it off for some time but after going through security, I realized
```
Comment (JCamp-12) [inline security/practical_security.md:2]: better to do it now vs. after you have data compromised

File: 1d-life/src/1d-life.js [hunk 3]
```diff
@@ -39,9 +39,18 @@
    * @param {*} x the current x coordinate in question
    */
   function getNewVal(lifeState, x) {
-    // !!! IMPLEMENT ME
-
-    return 0; // instead of this
+    let start = lifeState[x];
```
Comment (JCamp-12) [inline 1d-life/src/1d-life.js:42]: great answer, simple and works!  really nice!

**Review by JCamp-12 — COMMENTED** [review]
great job Pat!! especially setting up 2 factor and a random pw generator.  Good practice!

File: 1d-life/src/1d-life.js [hunk 4]
```diff
@@ -39,9 +39,18 @@
...
+    let right = lifeState[x + 1];
+    let sum = start + left + right;
+
+    if (x < 1 || x >= lifeState.length - 1) {
+      return 0;
+    } else if (sum === 0 || sum === 3) {
+      return 0;
+    } else {
+      return 1;
+    }
```
Comment (amoscato) [inline 1d-life/src/1d-life.js:53]: Very unique solution! I really like the approach you took here.

File: security/ThreatModel.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+After looking at our newly rolled out app for our loan service, it seems that personal
+user information is not getting stored properly. If I look at the code more closely, it 
+turns out that client info and passwords are exposed when they should be hashed at the very least. 
+Potential adversaries to this would be identify thiefs and quite frankly anyone looking to make a quick buck.
+I advise that the app be shut down until we can identify the root cause for the issue and fix the problem so 
+that user info is not continuing to be exposed.
+
+Furthermore we need to reach out to all users and have them reset their passwords and usernames to mitigatre the risk. 
+I would also recommend that we prompt user to install 2 factor authentication which further mitigates any
```
Comment (amoscato) [inline security/ThreatModel.md:17]: Absolutely! If you were to identify a threat like this, all users should be notified right away and made to change their passwords and enable any other security features until you can be sure that their data is safe.

File: security/practical_security.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+it was probably in my best interest to get it done to further protect myself. On top of that I 
+used a random password generator to help protect both my email and banking accounts to make
+that more secure. I made sure to store the password in a place that I could get to if I need it but that would be extremely difficult for anyone else to get to.
```
Comment (amoscato) [inline security/practical_security.md:5]: Awesome! Two factor and difficult passwords are super helpful in making sure that your information is secure. I recommend setting up a password manager as well. Most have features that will automatically generate secure passwords for you and change the passwords to your accounts automatically for you. It makes it super easy to be secure.

**Review by amoscato — COMMENTED** [review]
Great job here, Patrick! I like that you took a unique approach to the CA problem.

**Pull request closed without merging**